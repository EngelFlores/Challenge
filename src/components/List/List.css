.filters {
  display: grid;
  grid-template-columns: 4fr 1fr 1fr 1fr;
  gap: 50px;
  background-color: rgba(224, 222, 222, 0.35);
  padding: 15px;
  border-radius: 50px;
  margin: 20px;
}

.title {
  font-size: 25px;
  padding: 20px;
}

.list {
  padding: 20px;
  width: 100%;
}

