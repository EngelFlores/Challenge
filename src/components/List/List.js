import React, { useState } from 'react';
import { Row } from '../Row/Row'
import { SearchBar } from '../SearchBar/SearchBar'
import { Filter } from '../Filter/Filter'
import { Sort } from '../Sort/Sort'
import { Reset } from '../Reset/Reset'
import "./List.css"

import { getAllUsers, getByStatus, sortBy } from "../../api/usersApi"

const List = ({ data }) => {
  const [filteredList, setFilteredList] = useState(data);

  const filterList = (event) => {
    const search = event.target.value
    const filtered = data.filter(element => {
      return element.email.toLowerCase().indexOf(search.toLowerCase()) !== -1 || element.primaryGroup.toLowerCase().indexOf(search.toLowerCase()) !== -1
    })
    return setFilteredList(filtered)
  }

  const handleFilter = async (status) => {
    try {
      const { data } = (await getByStatus(status));
      setFilteredList(data)
    } catch (error) {
      throw error
    }
  }

  const handleSort = async (filter) => {
    try {
      const { data } = await sortBy(filter);
      setFilteredList(data)
    } catch (error) {
      throw error
    }
  }

  const handleResetFilters = async () => {
    const { data } = await getAllUsers();
    setFilteredList(data)
  }

  return (
    <>
      <h1 className="title">List of Users</h1>
      <div className="filters">
        <SearchBar placeholder="Email or Group" onChange={filterList}></SearchBar>
        <Filter title="Status" handleFilter={handleFilter}></Filter>
        <Sort title="Sort" handleSort={handleSort}></Sort>
        <Reset handleResetFilters={handleResetFilters}></Reset>
      </div>
      <div>
        <table className="list">
          <thead>
            <tr>
              <th>Email</th>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Primary Group</th>
              <th>Phone Number</th>
              <th>Hours Studied</th>
              <th>Status</th>
            </tr>
          </thead>
          <tbody data-testid="row">
            {filteredList.map(information => {
              return <Row key={information.id} information={information}></Row>
            })}
          </tbody>
        </table>
      </div>
    </>
  )
}

export { List }
