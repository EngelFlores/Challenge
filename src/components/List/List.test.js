import { render, waitFor, fireEvent } from '@testing-library/react';
import React from 'react'

import { List } from './List'

describe('Should render right information in the list', () => {
  const mock = [{
    "id": 53,
    "firstName": "Susanna",
    "lastName": "Legion",
    "email": "slegiona@prlog.org",
    "primaryGroup": "pretium",
    "hoursStudied": 2857,
    "phoneNumber": "152-210-9118",
    "status": "active"
  }, {
    "id": 352,
    "firstName": "Letta",
    "lastName": "Fryatt",
    "email": "lfryattb@furl.net",
    "primaryGroup": "imperdiet sapien",
    "hoursStudied": 1795,
    "phoneNumber": "543-217-4959",
    "status": "active"
  }]

  test('Render filtered information from user', async () => {
    const { getByTestId } = render(<List data={mock} />)
    const searchInput = await waitFor(() => getByTestId('search-input'))
    const list = await waitFor(() => getByTestId('row'))
    const value = "lfryattb@furl.net"
    fireEvent.change(searchInput, { target: { value } })

    expect(list).toHaveTextContent('Letta')
  })

  test('Component match snapshot', async () => {
    const { getByTestId } = render(<List data={mock} />)
    const list = await waitFor(() => getByTestId('row'))

    expect(list).toMatchSnapshot()
  })

})