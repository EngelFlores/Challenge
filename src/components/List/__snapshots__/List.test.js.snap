// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render right information in the list Component match snapshot 1`] = `
<tbody
  data-testid="row"
>
  <tr>
    <td
      data-testid="email"
    >
      slegiona@prlog.org
    </td>
    <td
      data-testid="first-name"
    >
      Susanna
    </td>
    <td
      data-testid="last-name"
    >
      Legion
    </td>
    <td
      data-testid="primary-group"
    >
      pretium
    </td>
    <td
      data-testid="phone-number"
    >
      152-210-9118
    </td>
    <td
      data-testid="hours-studied"
    >
      2857
    </td>
    <td
      data-testid="status"
    >
      active
    </td>
  </tr>
  <tr>
    <td
      data-testid="email"
    >
      lfryattb@furl.net
    </td>
    <td
      data-testid="first-name"
    >
      Letta
    </td>
    <td
      data-testid="last-name"
    >
      Fryatt
    </td>
    <td
      data-testid="primary-group"
    >
      imperdiet sapien
    </td>
    <td
      data-testid="phone-number"
    >
      543-217-4959
    </td>
    <td
      data-testid="hours-studied"
    >
      1795
    </td>
    <td
      data-testid="status"
    >
      active
    </td>
  </tr>
</tbody>
`;
